src/mqt/bench/benchmark_generator.py

```diff
@@ -317,13 +321,18 @@ def get_benchmark(
         benchmark_instance_name: Input selection for some benchmarks, namely "groundstate" and "shor"
         compiler: "qiskit" or "tket"
         CompilerSettings: Data class containing the respective compiler settings for the specified compiler (e.g., optimization level for Qiskit or placement for TKET)
-        gate_set_name: "ibm", "rigetti", "ionq", "oqc", or "quantinuum"
+        provider_name: "ibm", "rigetti", "ionq", "oqc", or "quantinuum"
         device_name: "ibm_washington", "ibm_montreal", "rigetti_aspen_m2", "ionq_harmony", "ionq_aria1", "oqc_lucy", "quantinuum_h2",
 
     Returns:
         Quantum Circuit Object representing the benchmark with the selected options, either as Qiskit::QuantumCircuit or Pytket::Circuit object (depending on the chosen compiler---while the algorithm level is always provided using Qiskit)
     """
 
+    if "gate_set_name" in kwargs:
+        msg = "gate_set_name is deprecated and will be removed in a future release. Use provider_name instead."
```

Comment:
Could you provide a test case to cover this message as well? Should be fairly easy an increased the overall coverage.